### PR TITLE
Fix log directory naming

### LINKED ISSUE #677: Log files are created under "smi" dir instead of application name

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Fix log directory naming for single-entrypoint app.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [x] Updated any relevant API documentation
- [x] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues
-   Fixes #677

### CODE REVIEW

**Comment by rkm** [comment]
Will need to take another look at this for the failing tests.

File: src/common/Smi.Common/Options/SmiCliInit.cs [hunk 1]
```diff
@@ -120,6 +117,16 @@ public static T Verify<T>(object parsedOptions)
             return asExpected;
         }
 
+        private static string GetHostProcessName(Type t)
+        {
+            string hostProcessName = t.Namespace?.Split('.')[1];
```
Comment (tznind) [inline src/common/Smi.Common/Options/SmiCliInit.cs:122]: Not sure but I think that `?.` will return null which you then index using [1].  So either drop the `?` or use `t.Namespace?.Split('.')?.Skip(1)?.FirstOrDefault()`

Comment (rkm) [reply src/common/Smi.Common/Options/SmiCliInit.cs:126]: This should work fine for all our existing apps (have tested manually), but the issue was the tests don't follow the same execution path which sets the HostProcessName elsewhere. Have fixed this now I think.

Comment (tznind) [reply src/common/Smi.Common/Options/SmiCliInit.cs:126]: Oh that's cool, didn't know that worked but seems it does

```
string s = null ;
string b = s?.Split('.')[1];
```
_Gives b==null_

**Review by tznind — APPROVED** [review]

**Pull request merged**